cpp/daal/include/algorithms/algorithm_container_base.h

```diff
@@ -65,6 +65,9 @@ class AlgorithmContainerIface
 class AlgorithmContainerIfaceImpl : public AlgorithmContainerIface
 {
 public:
+    AlgorithmContainerIfaceImpl()                                    = default;
```

Comment:
Is this one actually needed? Does it fail to compile without it?

Reply:
Yes, it fails to compile with this error:

```
In file included from ./__release_lnx/daal/latest/include/algorithms/algorithm_base.h:29:
./__release_lnx/daal/latest/include/algorithms/algorithm_container_base_batch.h:56:5: error: explicitly defaulted default constructor is implicitly deleted [-Werror,-Wdefaulted-function-deleted]
   56 |     AlgorithmContainer()                           = default;
      |     ^
./__release_lnx/daal/latest/include/algorithms/algorithm_container_base_batch.h:53:35: note: default constructor of 'AlgorithmContainer<daal::batch>' is implicitly deleted because base class 'AlgorithmContainerIfaceImpl' has no default constructor
   53 | class AlgorithmContainer<batch> : public AlgorithmContainerIfaceImpl
      |                                   ^
./__release_lnx/daal/latest/include/algorithms/algorithm_container_base_batch.h:56:54: note: replace 'default' with 'delete'
```

Reply:
Got it. But then, does deleting the other constructor have some effect if it's not called explicitly?

Reply:
It could have the effect of eliminating the "rule of three violation" hits in Coverity. I hope.
And it seems I have to mark copy assignment operator deleted as well.

Reply:
> And it seems I have to mark copy assignment operator deleted as well.

Yes, that was also highlighted by coverity. Hopefully that should do.